x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -199,8 +199,7 @@ booleanValue
     ;
 
 numericValue
-    : decimalValue
-    | integerValue
+    : (PLUS | MINUS)? (decimalValue | integerValue)
```

Comment:
Yes, that's a bit of a stretch due to the fact that `numericValue` was defined specifically for the constant square brackets scope, while `decimalValue` and and `integerValue` are used in a wider scope that already accepts signs in a completely different way (ie. via `operatorExpression`). Adding signs to them would make the grammar someway ambiguous

Reply:
Ah, so this is why `numericValue` is used inside `constant` as well - to enable arrays.

Maybe a short comment would help that explains why `numericValue` needs a sign but we otherwise do not attach signs directly to `integerValue` and `decimalValue`. Additionally, if `numericValue`'s usage is supposed to be restricted for arrays, maybe `numericArrayValue` as a name wouldn't would suggest as much generality.

Anyway, the PR is perfectly fine as-is, IMO, and my questions/nits come from the angle of someone who's just briefly familiar with the grammar here :)